src/main/scala/net/psforever/actors/zone/BuildingActor.scala

```diff
@@ -201,6 +205,22 @@ class BuildingActor(
         }
         Behaviors.same
 
+      case AmenityStateChange(terminal: CaptureTerminal, data) =>
+        // Notify amenities that listen for CC hack state changes, e.g. wall turrets to dismount seated players
+        building.Amenities.filter(x => x.isInstanceOf[CaptureTerminalAware]).foreach(amenity => {
+          amenity.Actor ! CaptureTerminalAwareBehavior.TerminalStatusChanged(terminal, data.get.asInstanceOf[Boolean])
+        })
```

Comment:
Please unwrap the optional value in `data` using a `match` statement.

Reply:
Amended